services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -195,6 +195,7 @@ public static void updateUserFromRep(UserModel user, UserRepresentation rep, Set
             user.setUsername(rep.getUsername());
         }
         if (rep.getEmail() != null) user.setEmail(rep.getEmail());
+        if (rep.getEmail() == "") user.setEmail(null);
```

Comment:
@ssilvert, what if you have spaces in the empty string ?

Reply:
@pedroigor That's actually a completely different issue.  You should never be allowed to set an email address to empty string.  But right now you can.  We don't check to see if the email has a valid format.

The thing this JIRA addresses is the case where you want to get rid of the email all together.  It used to save the address as empty string.  Then if you want to get rid of the email on another user it would try to save it as empty string and it would think that it was a duplicate email address.